### PR TITLE
Refactor Steward support

### PR DESCRIPTION
**Pull request by gauthierm**
- drop legacy Selenium code
 - organize project into more namespaces
 - add a DateEntry component
 - use league/uri for URL handling
 - use vlucas/dotenv for config management
 - use helper methods provided by steward and include those helpers in pages and components
 - use PSR-2 for CS
 - add tools for taking screenshots

TODO

 - [x] split Atom testing into a separate package or namespace
 - [x] finish up the config stuff
 - [x] clean up getNormalizedURL maybe move to shared trait or remove from URL assertion trait
 - [x] update CourseHost tests to use new APIs. See https://github.com/silverorange/course-host/pull/909 for using these updates with CourseHost.
 - [x] update Starmaker tests to use new APIs started in https://github.com/silverorange/starmaker/pull/342
 - [x] improve bootstrap script with default params and better env settings
 - [ ] update Primary Care RAP tests to use new APIs

### CODE REVIEW

File: src/Components/DateEntry.php [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+ * @license   http://www.gnu.org/copyleft/lesser.html LGPL License 2.1
+ */
+class DateEntry extends AbstractComponent
+{
+    // {{{ protected properties
+
+    /**
+     * @var array
+     */
+    protected static $month_map = array(
```
Comment (gauthierm) [inline src/Components/DateEntry.php]: square brackets

File: src/Console/Command/CleanCommand.php [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+    /**
+     * @param InputInterface $input
+     * @return array
+     */
+    protected function resolveConfiguration(InputInterface $input)
+    {
+        // Steward uses a steward.yml file for some config options. We override
+        // it here so we can specify config options in the environment and so
+        // we can dynamically configure the capabilities resolver class. All
```
Comment (gauthierm) [inline src/Console/Command/CleanCommand.php]: Not true for config

File: src/Fixture/ContentTrait.php [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+ * @package   Turing
+ * @copyright 2018 silverorange
+ */
+trait ContentTrait
+{
+    // {{{ protected function getContent()
+
+    protected function getContent($filename)
+    {
+        $path = getenv('CONTENT_PATH');
```
Comment (gauthierm) [inline src/Fixture/ContentTrait.php]: use ternary operator

File: src/Screenshot/ScreenshotTrait.php [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+        $element = null;
+
+        if (!$element instanceof WebDriverElement) {
+            $element = $this->findByTag('body');
+        }
+
+        $taker = new Taker($this->wd);
+        $screenshot = $taker->take($element, $padding);
+
+        $path = getenv('SCREENSHOTS_PATH');
```
Comment (gauthierm) [inline src/Screenshot/ScreenshotTrait.php]: use ternary operator

File: src/Selenium/CapabilitiesResolver.php [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+            getenv('SELENIUM_HEADLESS')
+        ) {
+            $chrome_options = new ChromeOptions();
+
+            $width = (int)(getenv('SELENIUM_WIDTH') ?: 800);
+            $height = (int)(getenv('SELENIUM_HEIGHT') ?: 600);
+
+            // In headless Chrome 60+, window size cannot be changed run-time:
+            // https://bugs.chromium.org/p/chromium/issues/detail?id=604324#c46
+            $chrome_options->addArguments(['--headless', '--window-size=' . $width . ',' . $height ]);
```
Comment (gauthierm) [inline src/Selenium/CapabilitiesResolver.php]: always set window size, optionally set headless

File: src/WebDriver/HasElementTrait.php [hunk 6]
```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace Silverorange\Turing\WebDriver;
+
+/**
+ * @package   Turing
+ * @copyright 2017-2018 silverorange
+ * @license   http://www.gnu.org/copyleft/lesser.html LGPL License 2.1
+ */
+trait HasElementTrait
```
Comment (charleswaddell) [inline src/WebDriver/HasElementTrait.php]: Is it just me or is it weird that this trait appears to wholly depend on another trait. Any reason we just don't combine them?

Comment (gauthierm) [reply src/WebDriver/HasElementTrait.php]: This is so we can use the trait in both `Component` and the base `AbstractTest`.

Comment (charleswaddell) [reply src/WebDriver/HasElementTrait.php]: Got it.

**Pull request merged**